spec/models/pb_core_spec.rb

```diff
@@ -0,0 +1,154 @@
+        title: 'SERIES; PROGRAM; ASSET',
+        duration: '01:23:45',
+        asset_type: 'Original footage',
+        series_description: 'SERIES-DESCRIPTION',
+        program_description: 'PROGRAM-DESCRIPTION',
+        asset_description: 'ASSET-DESCRIPTION',
+        id: 'V_5FDB1545443B427888C90E7B15F3783A',
+        thumbnail_src: "#{base}/asset_thumbnails/#{id}.jpg",
+        proxy_srcs: ['mp4', 'webm'].map{|ext| "#{base}/asset_proxies/#{id}.#{ext}"},
+        rights_summary: 'RIGHTS-SUMMARY',
+        contributors: [
+          PBCoreNameRole.new('contributor', 'CONTRIBUTOR-NAME-1', 'CONTRIBUTOR-ROLE-1'),
+          PBCoreNameRole.new('contributor', 'CONTRIBUTOR-NAME-2', 'CONTRIBUTOR-ROLE-2')],
+        creators: [
+          PBCoreNameRole.new('creator', 'CREATOR-NAME-1', 'CREATOR-ROLE-1'),
+          PBCoreNameRole.new('creator', 'CREATOR-NAME-2', 'CREATOR-ROLE-2')],
+        publishers: ['PUBLISHER-1', 'PUBLISHER-2'],
+        media_type: 'Video',
+        video?: true,
+        audio?: false,
+        image?: false,
+        subjects: ['SUBJECT-1', 'SUBJECT-2'],
+        genres: ['GENRE-1', 'GENRE-2'],
+        topics: ['TOPIC-1', 'TOPIC-2'],
+        locations: ['LOCATION-1', 'LOCATION-2'],
+        us_only?: true,
+        password_required?: true,
+        special_collections: ['war_peace'],
+        special_collection_tags: ['war_interview'],
+        scholar_exhibits: ['needlework'],
+        aapb_url: 'http://americanarchive.org/',
+        boston_tv_news_url: nil,
+        extensions: ["mp4", "webm"],
+        outside_url: 'http://americanarchive.org/',
+        transcript_srcs: ['https://s3.amazonaws.com/openvault.wgbh.org/catalog/asset_transcripts/V_5FDB1545443B427888C90E7B15F3783A.xml']}
+      assertions[:to_solr] = assertions.slice(
+        :id, :title, :thumbnail_src, :year, :series_title, :program_title,
+        :subjects, :locations, :genres, :topics, :asset_type, :media_type, 
+        :scholar_exhibits, :special_collections, :special_collection_tags)
+      .merge({
+          xml: pbc_xml,
+          text: ["SERIES", "PROGRAM", "PROGRAM-NUMBER", "ASSET", "12/31/1999", 
+            "1999", "SERIES; PROGRAM; ASSET", "SERIES-DESCRIPTION", 
+            "PROGRAM-DESCRIPTION", "ASSET-DESCRIPTION", "CONTRIBUTOR-NAME-1", 
+            "CONTRIBUTOR-ROLE-1", "CONTRIBUTOR-NAME-2", "CONTRIBUTOR-ROLE-2", 
+            "CREATOR-NAME-1", "CREATOR-ROLE-1", "CREATOR-NAME-2", "CREATOR-ROLE-2", 
+            "PUBLISHER-1", "PUBLISHER-2", "SUBJECT-1", "SUBJECT-2", "LOCATION-1", 
+            "LOCATION-2", "GENRE-1", "GENRE-2", "TOPIC-1", "TOPIC-2", "RIGHTS-SUMMARY"]
+        })
+
+      pbc = PBCore.new(pbc_xml)
+
+      assertions.each do |method, value|
+        it "\##{method} method works" do
+          expect(pbc.send(method)).to eq(value)
+        end
+      end
+
+      it 'tests everthing' do
+        expect(assertions.keys.sort).to eq(PBCore.instance_methods(false).sort)
+      end
+    end
+  end
+end
```

Comment:
@mccalluc you should be careful when creating variables within an RSpec context or example. When you are in a context (i.e. a `describe` block), the value of `self` is a class that extends `RSpec::ExampleGroup`, ad when you are in an example (i.e. an `it` block), teh value of `self` is an instance of that class.

RSpec will allow you to appear to set local variables without complaining. But I think this may be bad, because you may accidentally overwrite an object that may be used elsewhere, resulting in undesirable rspec behavior.

For instance, there is a class methods called `metadata` in the `RSpec::ExampleGroup` class... so you can do this...

```ruby
```